File: python/greedy/hand-of-straights.py
class Solution:
    def isNStraightHand(self, hand: List[int], groupSize: int) -> bool:
        count = defaultdict(int)
        for i in hand:
            count[i] += 1

        for i in sorted(count.keys()):
            while count[i] > 0:
                for j in range(groupSize):
                    count[i + j] -= 1
                    if count[i + j] < 0:
                        return False

        return True

File: c++/greedy/hand-of-straights.cc
class Solution {
public:
    bool isNStraightHand(vector<int>& hand, int groupSize) {
        std::unordered_map<int, int> count;
        for (int card : hand) {
            count[card]++;
        }

        std::sort(hand.begin(), hand.end());
        for (int i = 0; i < hand.size(); ++i) {
            if (count[hand[i]] == 0) {
                continue;
            }
            int update = count[hand[i]];

            for (int j = hand[i]; j < hand[i] + groupSize; ++j) {
                count[j] -= update;
                if (count[j] < 0) {
                    return false;
                }
            }
        }
        return true;
    }
};
